stores/metadata_test.go

```diff
@@ -4307,57 +4305,94 @@ func TestTypeCurrency(t *testing.T) {
 	ss := newTestSQLStore(t, defaultTestSQLStoreConfig)
 	defer ss.Close()
 
+	// prepare the table
+	if isSQLite(ss.db) {
+		if err := ss.db.Exec("CREATE TABLE currencies (id INTEGER PRIMARY KEY AUTOINCREMENT,c BLOB);").Error; err != nil {
+			t.Fatal(err)
+		}
+	} else {
+		if err := ss.db.Exec("CREATE TABLE currencies (id INT AUTO_INCREMENT PRIMARY KEY, c BLOB);").Error; err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	// insert currencies in random order
+	values := []interface{}{
+		bCurrency(types.ZeroCurrency),
+		bCurrency(types.NewCurrency64(1)),
+		bCurrency(types.MaxCurrency),
+	}
+	frand.Shuffle(len(values), func(i, j int) { values[i], values[j] = values[j], values[i] })
```

Comment:
Imo the randomness introduced by the shuffle is more dangerous than helpful because it leaves a >10% chance for broken code to pass the test.
If you insert the values in a non-sorted order and they are returned in a sorted way, that should be pretty solid confirmation.